### PR TITLE
[plugin.video.iwm] 1.0.0

### PR DESCRIPTION
**Pull request by FraserChapman**
### Description

This add-on enables playing of videos and movies from the Imperial War Museum Film website https://film.iwmcollections.org.uk/

This add-on is not created, maintained or in any way affiliated with the Imperial War Museum.
It only provides an interface to access the free content on the IWM Film website from Kodi.

I realise it isn't going to be Python 3 compatible yet :(

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by FraserChapman** [comment]
doh!

**Comment by FraserChapman** [comment]
Ah just noticed a bug with the cache/recent clearance...

**Comment by FraserChapman** [comment]
@Rechi - ah have to shelve this for now then as AFAIK there are no mac builds with a working Python 3 interpreter (see https://github.com/xbmc/xbmc/issues/16256) - OOI what is failing? Presume it is a dependancy rather than any code in the add-on itself? Would love to update and get this and my other addons updated but without a working build for P3 on mac I'm a bit stumped.

**Comment by FraserChapman** [comment]
Think I've cracked it - ran a VM with Unbuntu 16 using the PPA at: https://launchpad.net/~wsnipex/+archive/ubuntu/kodi-python3/ - the add-on seems to run AOK with the small changes made - fingers crossed it is now P3 compatible

File: plugin.video.iwm/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+__pycache__/
```
Comment (Rechi) [inline plugin.video.iwm/.gitignore]: drop this file from the PR

File: plugin.video.iwm/resources/lib/kodiutils.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def set_setting(setting, value):
+    # type (str, Any) -> None
+    ADDON.setSetting(setting, str(value))
+
+
+def get_setting_as_bool(setting):
+    # type (str) -> bool
+    return get_setting(setting).lower() == "true"
```
Comment (Rechi) [inline plugin.video.iwm/resources/lib/kodiutils.py]: `ADDON.getSettingBool(setting)`

File: plugin.video.iwm/resources/lib/kodiutils.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def get_setting_as_float(setting):
+    # type (str) -> float
+    try:
+        return float(get_setting(setting))
```
Comment (Rechi) [inline plugin.video.iwm/resources/lib/kodiutils.py]: `ADDON.getSettingNumber(setting)`

File: plugin.video.iwm/resources/lib/kodiutils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    except ValueError:
+        return 0
+
+
+def get_setting_as_int(setting):
+    # type (str) -> int
+    try:
+        return int(get_setting_as_float(setting))
```
Comment (Rechi) [inline plugin.video.iwm/resources/lib/kodiutils.py]: `ADDON.getSettingInt(setting)`

**Comment by FraserChapman** [comment]
@Rechi - Thanks so much for that - if that is it then I'm certain I can update my other add-ons to make them compatible too. Cheers

File: plugin.video.iwm/.gitignore (hunk 1 above)
Comment (Rechi) [reply plugin.video.iwm/.gitignore]: the whole .gitignore file not just this line

Comment (FraserChapman) [reply plugin.video.iwm/.gitignore]: Ah my bad :)

**Comment by FraserChapman** [comment]
Woot :)

**Pull request merged**